src/python/pants/backend/python/rules/pex.py

```diff
@@ -4,23 +4,30 @@
 from pants.backend.python.rules.download_pex_bin import DownloadedPexBin
 from pants.backend.python.subsystems.python_native_code import PexBuildEnvironment, PythonNativeCode
 from pants.backend.python.subsystems.python_setup import PythonSetup
-from pants.engine.fs import Digest
+from pants.engine.fs import EMPTY_DIRECTORY_DIGEST, Digest, DirectoriesToMerge
 from pants.engine.isolated_process import (ExecuteProcessRequest, ExecuteProcessResult,
                                            MultiPlatformExecuteProcessRequest)
 from pants.engine.platform import Platform, PlatformConstraint
 from pants.engine.rules import optionable_rule, rule
 from pants.engine.selectors import Get
-from pants.util.objects import datatype, hashable_string_list, string_optional, string_type
+from pants.util.objects import Exactly, datatype, hashable_string_list, string_optional, string_type
 from pants.util.strutil import create_path_env_var
 
 
-class RequirementsPexRequest(datatype([
+class MakePexRequest(datatype([
   ('output_filename', string_type),
   ('requirements', hashable_string_list),
   ('interpreter_constraints', hashable_string_list),
   ('entry_point', string_optional),
+  ('input_files_digest', Exactly(Digest, type(None))),
```

Comment:
Rather that "either Digest or None", you can require a digest, and default to `EMPTY_DIRECTORY_DIGEST` (aka "the null object pattern").